posthog/queries/retention/actors_query.py

```diff
@@ -62,7 +62,10 @@ def actors(self):
         )
 
         results = insight_sync_execute(
-            actor_query, query_type="retention_actors", filter=self._filter, args=actor_query_params
+            actor_query,
+            {**actor_query_params, **self._filter.hogql_context.values},
```

Comment:
Why isn't this done by actor query itself?

Problem is - writing code you now need to know that _sometimes_ you need to inject some params in from somewhere but other times you do not.

Reply:
It was hard to understand how all these queries work --> which of these get inlined as subqueries, which are full queries. This means I was never sure what's the *right time* to inline the hogql values into the params. Hence I took an alternative approach: the entire HogQL value gathering is something that is expected to be controlled outside the queries. I don't mind re-evaluating this approach, as the worse case for that is just passing a lot of duplicate (yet still unique) params around.

Reply:
Fair - I can get why we got here. We'll need to revisit this soonish and push these params 'deeper' again (it's a maintanence nightmare otherwise) but not sure we should block this PR further.

Reply:
I'm happy to take on cleaning up all these queries in a followup... or even here if you think it's blocking, though I'd obviously prefer to do a followup. For now, the worst thing that can happen in a future query that forgets to add these values... is a missing `%(hogql_val_1)s` error. 

It seems very possible to find a clear boundary between queries and subqueries, and inline those values into queries there. Those big query classes like `Retention`, `ClickhouseFunnelTrends`, etc might be just that boundary... but then it's not predefined where their `Filter`-s come from... blergh... 🤔